src/Internal/Types/Interval.purs

```diff
@@ -861,6 +862,25 @@ derive instance Eq PosixTimeToSlotError
 instance Show PosixTimeToSlotError where
   show = genericShow
 
+explainPosixTimeToSlotError :: PosixTimeToSlotError -> String
+explainPosixTimeToSlotError = case _ of
+  CannotFindTimeInEraSummaries atime ->
+    "Time " <> show atime <> " is not in era summaries."
+  PosixTimeBeforeSystemStart t ->
+    "Time " <> show t <> " is before the system start."
+  StartTimeGreaterThanTime atime ->
+    "Time " <> show atime <> " is before the era start time."
+  EndSlotLessThanSlotOrModNonZero slot mtime ->
+    "Time is after the era end time. Calculated slot: " <> show slot
+      <> ", calculated residual time after end: "
+      <> show mtime
+  CannotGetBigIntFromNumber' ->
+    "Internal error: cannot convert a time to a usable value. " <>
+      "This should be impossible."
+  CannotGetBigNumFromBigInt' ->
+    "Internal error: cannot convert a slot number to a usable value. " <>
+      "This should be impossible."
```

Comment:
Let's get rid of these error variants. We should not provide a way to dispatch on them, because they are conversion errors that we believe should not happen.

We can even remove the possibility of this even theoretically happening.

`relSlotFromRelTime` uses conversion to Number before dividing and module, we should remove it and just use `div` and `mod` in BigInt. Please do this refactoring.

Reply:
It seems it's a bit trickier than this. Right from the gate, we have:

```haskell
  -- Get POSIX time for system start:
  sysStartPosix <- liftM CannotGetBigIntFromNumber' $ sysStartUnixTime sysStart
```

The `liftM` is required there, because `sysStartUnixTime` calls `BigInt.fromNumber`, which _must_ return `Maybe`, as `Number` can be `NaN` or an infinity. We can potentially 'escape' this by making the whole thing partial, or being in a monad where we can throw exceptions. I'd argue that this is literally impossible (because system start time being a NaN or infinity should never occur), but I'm not quite sure how to proceed here.

Reply:
Use `unsafePartial <<< fromJust`. We can rely on the fact that system start is not a NaN or Infinity.

Reply:
Just make this assumption explicit in a comment

Reply:
^